main/src/library/MutList.flix

```diff
@@ -634,15 +580,13 @@ mod MutList {
     /// If the given index `i` exceeds the length of the mutable list, the element is inserted at the last position.
     ///
     pub def insert!(x: a, i: Int32, v: MutList[a, r]): Unit \ r =
-        let MutList(r, a, l) = v;
-        let len = Ref.get(l);
-        if (capacity(v) - len == 0)
-            reserve!(len, v)
+        if (capacity(v) - length(v) == 0)
+            reserve!(length(v), v)
         else ();
-        let sub = Array.copyOfRange(r, i, len, Ref.get(a));
-        Array.updateSequence!(i + 1, sub, Ref.get(a));
-        Array.put(x, i, Ref.get(a));
-        Ref.put(len + 1, l)
+        let sub = Array.copyOfRange(v->r, i, length(v), v->values);
+        Array.updateSequence!(i + 1, sub, v->values);
+        Array.put(x, i, v->values);
+        v->length = length(v) + 1
```

Comment:
I would replace length(v) with v->length everywhere. I think of the length function as part of the API. There is no reason not to access the struct field directly. In particular this line shows that it would be more clear to access the field directly.